webapp/src/main/java/com/box/l10n/mojito/service/tm/TMTextUnitStatisticService.java

```diff
@@ -198,7 +200,22 @@ private void uploadStatistics(Map <ImportTextUnitStatisticsBody, List<TMTextUnit
                                 return statistic;
                             });
                 })
-                .collect(Collectors.toList());
+                .collect(Collectors.collectingAndThen(
+                        Collectors.toMap(TMTextUnitStatistic::getTMTextUnit, Function.identity(), (stat1, stat2) -> {
+                            stat1.setLastDayUsageCount(stat1.getLastDayUsageCount() + stat2.getLastDayUsageCount());
+                            stat1.setLastPeriodUsageCount(stat1.getLastPeriodUsageCount() + stat2.getLastPeriodUsageCount());
+                            DateTime stat1date = stat1.getLastSeenDate();
+                            DateTime stat2date = stat2.getLastSeenDate();
+                            DateTime maxDate = stat1date.compareTo(stat2date) >= 0 ? stat1date : stat2date;
+                            stat1.setLastSeenDate(maxDate);
+
+                            logger.warn("Merged statistics information for TextUnit with id: {} , name: {}",
```

Comment:
warn? is this something that is not expected to happen regularly?

Reply:
_Ideally_ we wouldn't want this to happen regularly - as the hope would be that the statistics data being sent in maps 1:1 or 1:N to units - without having two _different_ statistics mapping back to the same TextUnit.
Emitting a warning as this implies the data could be a bit less accurate for those units - and would allow us to trace it back if ever needed.